### ISSUE TITLE
Eval bug: Gemma 4 31B finetunes crash

### ISSUE THREAD

**Issue by frenzybiscuit** [ggml-org/llama.cpp#22483]
### Name and Version

/llama.cpp-b8953/build/bin$ ./llama-server --version
ggml_cuda_init: found 4 CUDA devices (Total VRAM: 96504 MiB):
  Device 0: NVIDIA GeForce RTX 3090, compute capability 8.6, VMM: yes, VRAM: 24126 MiB
  Device 1: NVIDIA GeForce RTX 3090, compute capability 8.6, VMM: yes, VRAM: 24126 MiB
  Device 2: NVIDIA GeForce RTX 3090, compute capability 8.6, VMM: yes, VRAM: 24126 MiB
  Device 3: NVIDIA GeForce RTX 3090, compute capability 8.6, VMM: yes, VRAM: 24126 MiB
version: 0 (unknown)
built with GNU 14.3.0 for Linux x86_64

### Operating systems

Linux

### GGML backends

CUDA

### Hardware

Ryzen 7590x + 128GB RAM + 4x3090

### Models

https://huggingface.co/BeaverAI/Artemis-31B-v1h-GGUF/resolve/main/Artemis-31B-v1h-Q8_0.gguf

### Problem description & steps to reproduce

Continue to have crash after crash when using sm tensor and parallel 2.

I've tried full reinstalls. I've tried using different CUDA versions. I've tried using different NCCL versions. I've tried different distros. I've tried different llamacpp versions.

### Relevant log output

slot get_availabl: id  1 | task -1 | selected slot by LCP similarity, sim_best = 1.000 (> 0.100 thold), f_keep = 0.891
slot launch_slot_: id  1 | task -1 | sampler chain: logits -> penalties -> ?dry -> ?top-n-sigma -> top-k -> ?typical -> top-p -> min-p -> ?xtc -> ?temp-ext -> dist
slot launch_slot_: id  1 | task 84957 | processing task, is_child = 0
slot update_slots: id  1 | task 84957 | new prompt, n_ctx_slot = 100096, n_keep = 0, task.n_tokens = 18665
slot update_slots: id  1 | task 84957 | n_past = 18665, slot.prompt.tokens.size() = 20958, seq_id = 1, pos_min = 19422, n_swa = 1024
slot update_slots: id  1 | task 84957 | Checking checkpoint with [17125, 18660] against 17641...
slot update_slots: id  1 | task 84957 | restored context checkpoint (pos_min = 17125, pos_max = 18660, n_tokens = 18661, n_past = 18660, size = 1200.019 MiB)
slot update_slots: id  1 | task 84957 | n_tokens = 18660, memory_seq_rm [18660, end)
slot update_slots: id  1 | task 84957 | prompt processing progress, n_tokens = 18661, batch.n_tokens = 1, progress = 0.999786
slot update_slots: id  1 | task 84957 | n_tokens = 18661, memory_seq_rm [18661, end)
slot init_sampler: id  1 | task 84957 | init sampler, took 2.57 ms, tokens: text = 18665, total = 18665
slot update_slots: id  1 | task 84957 | prompt processing done, n_tokens = 18665, batch.n_tokens = 4
srv  log_server_r: done request: POST /v1/chat/completions 127.0.0.1 200
srv  params_from_: Chat format: peg-gemma4
slot get_availabl: id  0 | task -1 | selected slot by LRU, t_last = 28403228503
srv  get_availabl: updating prompt cache
srv   prompt_save:  - saving prompt with length 45061, total state size = 4720.925 MiB
./artemis.sh: line 2: 52571 Killed                     llama.cpp-b8953/build/bin/llama-server --model gguf/Artemis-31B-v1h-Q8_0.gguf --api-key keygoeshere --host 127.0.0.1 --port 8080 -sm tensor --cache-ram 65536 --ctx-size 200000 --parallel 2

**Comment by Michelle-mhy** [comment]
我也遇到了，我用的是 D:\llama-b8967-bin-win-vulkan-x64> .\llama-server.exe -m "D:\vlm\gemma4\gemma-4-E2B-it-Q4_K_M.gguf" --port 8081 -c 4096 --mmproj "D:\vlm\gemma4\mmproj-BF16.gguf" --verbose

**Comment by ddh0** [comment]
Hi, if you could please post your actual crash logs so I could try to help you - I am looking into some issues with Gemma 31B too. At least it might help narrow down what exactly is happening.
